### ISSUE TITLE
After change advertising esp_ble_adv_data_t Iphone cannot connect (IDFGH-6097)

### ISSUE THREAD

**Issue by rolan82** [espressif/esp-idf#7776]
## Environment
Module or chip used: ESP32-WROOM-32U
IDF version: v4.2
Build System: idf.py
Compiler version : xtensa-esp32-elf-gcc (crosstool-NG esp-2020r3) 8.4.0
Operating System: Linux
Using an IDE?: VSCode
Power Supply: dc/dc 3.3V
Iphone: X Max , v.14.7.1

## Problem Description
My code is based on the gatt_security_server example. Iphone must be connected to our device. I have 2 firmwares. In the second firmware I changed `include_name = true ` in the `esp_ble_adv_data_t adv_data` structure.
After I update the firmware of our device with firmware from v.1 to firmware v.2, Iphone cannot connect to our device. For tests, I use nRF Connect.

When I press "Connect" to our device in nRF, the popup "peer remove pairing information" appears. The connection is not running. Until the user manually deletes the pair in the system settings of the bluetooth phone.
It is unclear why the phone thinks that the information about the pair has been deleted.
This problem is only with the Iphone. There is no such problem with Android

### Expected Behavior
Changing the firmware should not delete pair information.

### Steps to reproduce
1. Create a pair and connect to a device with firmware v.1
2. Reflash the device with firmware  v2 (without deleting the pair in the device or in the phone)
3. Try to connect the Iphone to the device

**Comment by xiewenxiang** [comment]
When pairing for the first time use firmware-1, mobile phones and module will saved the bonding information locally, after re-flash the firmware-2, the previous bonding information will be deleted, but the phone side still have it.
The module is still the same module, device address has not changed, The phone thinks it's a previous device, If it still use the previous bonding information to pair will module, the pairing will fail. So, you need to delete the local bonding information in mobile phone.

**Comment by rolan82** [comment]
@xiewenxiang Thanks for reply!
Maybe I wrote it incorrectly, but I don't "re-flash". I did an ota update.
In this case, the pair information should not have been lost in the module.

>  after re-flash the firmware-2, the previous bonding information will be deleted

It is not clear to me in what cases the module deletes information about a pair. Can you answer more broadly?

**Comment by zhp0406** [comment]
Have you resolved your issue?

**Comment by rolan82** [comment]
> Have you resolved your issue?

No. The user must delete information about the Bluetooth pair in the phone
